pkg/handler/processor/processor.go

```diff
@@ -74,3 +74,10 @@ type SourceInformation struct {
 	// Source describes the source which the collector got this information
 	Source string
 }
+
+// Emitter processes a document
+type Emitter func(*Document) error
```

Comment:
Should `Emitter` and `ErrHandler` not be defined in the collector? as the Collector is the one emitting to the processor. I dont see the reason for it to be here.